docs/builtins.rst

```diff
@@ -0,0 +1,217 @@
+    Accepts the same parameters as `Intl.PluralRules`_ API.
+
+    Example::
+  
+      liked-count = { PLURAL($num) ->
+        [one]   One person liked your message
+       *[other] { $num } people liked your message
+      }
+      
+
+``LIST``
+    Formats a list to a string in a given locale.
+    Accepts the same parameters as `Intl.ListFormat`_ API.
+
+    Example::
+
+      users = { LIST($user1, $user2, $user3) }
+
+      users2 = { LIST($users) }
+
+
+``LEN``
+    Returns the number that represents the length of the list argument.
+    Similar to JS `Array.length`_ API.
+
+    Example::
+
+      unread-emails = Number of unread emails: { LEN($emails) }
+
+``TAKE``
+    Returns a slice of a list.
+    Similar to JS `Array.prototype.slice`_ API.
+
+    Example::
+
+      first-user = Primarely, { TAKE($users, 1) } likesd your message.
+
+``DROP``
+    Returns a slice of the list starting from a given index.
+    Similar to JS `Array.prototype.slice`_ API.
+
+    Example::
+
+      more-users = But ultimately, { DROP($users, 1) } like your message as well.
+
+
+Implicit use
+============
+
+In order to simplify most common scenarios, L20n will run a default
+built-in while resolving placeables.
+
+For the list of implicit built-ins, the implict example has exactly the same
+result as the explicit one.
+
+``NUMBER``
+    If the variable passed from the developer is a number and is used in
+    a placeable, L20n will implicitly call a `NUMBER` built-in on it.
+
+    Example::
+
+      emails = Number of unread emails { $unreadEmails }
+
+      emails2 = Number of unread emails { NUMBER($undeadEmails) }
+
+``DATETIME``
+    If the variable passed from the developer is a date and is used in
+    a placeable, L20n will implicitly call a `DATE` built-in on it.
+
+    Example::
+
+      log-time = Entry time: { $date }
+
+      log-time2 = Entry time: { DATETIME($date) }
+
+``PLURAL``
+    If the variable passed from the developer is a number and is used in
+    a selector expression, L20n will implicitly call a `PLURAL` built-in on it.
+
+    Example::
+
+      liked-count = { PLURAL($num) ->
+        [one]   One person liked your message
+       *[other] { $num } people liked your message
+      }
+
+      liked-count2 = { $num ->
+        [one]   One person liked your message
+       *[other] { $num } people liked your message
+      }
+
+``LIST``
+    If the variable passed from the developer is a number and is used in
+    a placeable, L20n will implicitly call a `LIST` built-in on it.
+
+    Also, if the placeable is a list of variables, L20n will implicitly
+    call a `LIST` built-in on it.
+
+    Example::
+
+      users = { LIST($user1, $user2, $user3) }
+
+      users2 = { $user1, $user2, $user3 }
+
+      users = { LIST($users) }
+
+      users2 = { $users }
+
+
+Partially resolved built-ins
+============================
+
+In the future we'll want to allow for partially resolved builtins to be
```

Comment:
That's already possible right now! https://github.com/l20n/l20n.js/blob/3ba0b9ad8f20a9540eb7ef6c4bda4604af4677e1/src/intl/polyfill.js#L6-L7